tests/test_plugins.py

```diff
@@ -1,46 +1,54 @@
-from cms.api import add_plugin, create_page
+from cms.api import add_plugin, create_page, create_title
 from cms.test_utils.testcases import CMSTestCase
 
-from djangocms_snippet.models import Snippet
+from .utils.factories import (
+    SnippetWithVersionFactory,
+)
 
 
 class SnippetPluginsTestCase(CMSTestCase):
 
     def setUp(self):
         self.language = "en"
+        self.superuser = self.get_superuser()
+
         self.home = create_page(
-            title="home",
-            template="page.html",
-            language=self.language,
+            title="home", template="page.html",
+            language=self.language, created_by=self.superuser,
         )
+        title_data = {
+            "title": "home", "template": "page.html", "language": self.language,
+            "created_by": self.superuser, "page": self.home,
+        }
+        self.home_pagecontent = create_title(**title_data)
```

Comment:
There's no need to create the title, it's done as part of the create_page api: https://github.com/django-cms/django-cms/blob/release/4.0.x/cms/api.py#L178

Alias has some good examples of using the api with versioning in the tests: https://github.com/divio/djangocms-alias/blob/7d90b7763278ff74ebe49f70420ecb9f0e2dc4c6/tests/base.py#L58